nntrainer/tensor/tensor.h

```diff
@@ -1923,6 +1924,48 @@ class Tensor {
     return continuous;
   }
 
+  /**
+   * @brief Check if data is valid number
+   */
+  bool checkDataValidation(bool print = false) const {
+    bool ret = true;
+
+    // It only support FP16 and FP32
+    if (getDataType() != Tdatatype::FP32 && getDataType() != Tdatatype::FP16)
+      return true;
+
+    std::string data;
+    for (unsigned int i = 0, len = size(); i < len; ++i) {
```

Comment:
not this pr, but later we can accelerate with SIMD or OMP.

Reply:
SIMD support is really necessary!! it's too slow.

Reply:
https://github.com/VcDevel/std-simd

or

something like:
```
#include <iostream>
#include <experimental/simd>

using intV = std::experimental::fixed_size_simd<int, 8>;
using floatV = std::experimental::fixed_size_simd<float, 11>;


int main()
{
    int ia[] = {3, 1, -14, 1, 5, 9, -14, 9};
    float fa[] = {0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0};
    intV iv {[ia](const auto i) { return ia[i]; }};
    floatV fv {[fa](const auto i) { return fa[i]; }};

    iv += 100;
    fv += 10.0f;

    std::cout<<iv[0]<<std::endl;
    std::cout<<fv[0]<<std::endl;
}

```

Reply:
I'll investigate this in later PR.